blockchain/chain.go

```diff
@@ -2118,6 +2126,11 @@ func New(config *Config) (*BlockChain, error) {
 		return nil, AssertError("blockchain.New excessive block size set lower than LegacyBlockSize")
 	}
 
+	// Default to TempDir if not set.
+	if config.FastSyncDataDir == "" {
+		config.FastSyncDataDir = os.TempDir()
+	}
```

Comment:
Is there a reason for using `os.TempDir()`? I would usually use `ioutil.TempDir()` which is more careful about checking for existence and also ensure uniqueness. Or just slap a uuid4 onto `os.TempDIR()`.

Reply:
On a separate note, `config.FastSyncDataDir` and `config.FastSync` are the only config values mutated here. Would it be better to leave the config as it was received and only set those values on new structure being created?

Reply:
We could do that, honestly I think this can just be removed as the only blockchain.New object that needs it has it explicitly passed in. I think removal of this is the best option.

Reply:
This has been removed. =)